In [2]:
!pip install mistralai

  Using cached mistralai-1.8.2-py3-none-any.whl.metadata (33 kB)
  Using cached eval_type_backport-0.2.2-py3-none-any.whl.metadata (2.2 kB)
Using cached mistralai-1.8.2-py3-none-any.whl (374 kB)
Using cached eval_type_backport-0.2.2-py3-none-any.whl (5.8 kB)


In [3]:
from mistralai import Mistral
import os
from dotenv import load_dotenv

load_dotenv()

True

In [4]:
api_key = os.environ["MISTRAL_API_KEY"]

client = Mistral(api_key=api_key)

In [5]:
uploaded_pdf = client.files.upload(
    file={
        "file_name": "AASZX2206288.pdf",
        "content": open("AASZX2206288.pdf", "rb"),
    },
    purpose="ocr"
)  

print(uploaded_pdf)

id='9e9c1fb6-2296-4bce-bd29-5235ee684906' object='file' size_bytes=256219 created_at=1750302212 filename='AASZX2206288.pdf' purpose='ocr' sample_type='ocr_input' source='upload' num_lines=0


In [6]:
retrieved_file = client.files.retrieve(file_id=uploaded_pdf.id)

In [8]:
signed_url = client.files.get_signed_url(file_id=uploaded_pdf.id)

In [10]:
ocr_response = client.ocr.process(
    model="mistral-ocr-latest",
    document={
        "type": "document_url",
        "document_url": signed_url.url,
    },
    include_image_base64=True
)

In [11]:
ocr_response

OCRResponse(pages=[OCRPageObject(index=0, markdown="# 921-41740101\n\n**AASZX2206288**\n\n**SHANGHAI TONGRUO CLOTHING CO., LTD**\n\n**BUILDING 9, NO.199, CHANGJIAN ROAD, BAOSHAN**\n\n**DISTRICT, SHANGHAI, CHINA**\n\n**CHRISTINE HUANG 13501692262**\n\n**AWET**\n\nAs the Carrier\n\nCopies 1, 2 and 3 of this Air Waybill are originals and have the same validity.\n\n**Consignee's Name and Address**\n\n**TA3 SWIM (FUTURE WOW, INC)**\n\n**2029 N BERENDO ST**\n\n**LOS ANGELES, CA 90027**\n\n**Consignee's Account Number**\n\n**A13**\n\n**2029 N BERENDO ST**\n\n**LOS ANGELES, CA 90027**\n\n**A13**\n\n**2029 N BERENDO ST**\n\n**LOS ANGELES, CA 90027**\n\n**Consignee's Account Number**\n\n**A13**\n\n**2029 N BERENDO ST**\n\n**LOS ANGELES, CA 90027**\n\n**Accounting Information**\n\n**FREIGHT COLLECT**\n\n|  IATA Code | Account No.  |\n| --- | --- |\n|  |   |\n\n**Place/Airport of Departure and requested Routing**\n\n**SHENZHEN, CHINA**\n\n|  IATA Code | Routing and Destination  |\n| --- | --- |\n|

In [12]:
print(ocr_response)

pages=[OCRPageObject(index=0, markdown="# 921-41740101\n\n**AASZX2206288**\n\n**SHANGHAI TONGRUO CLOTHING CO., LTD**\n\n**BUILDING 9, NO.199, CHANGJIAN ROAD, BAOSHAN**\n\n**DISTRICT, SHANGHAI, CHINA**\n\n**CHRISTINE HUANG 13501692262**\n\n**AWET**\n\nAs the Carrier\n\nCopies 1, 2 and 3 of this Air Waybill are originals and have the same validity.\n\n**Consignee's Name and Address**\n\n**TA3 SWIM (FUTURE WOW, INC)**\n\n**2029 N BERENDO ST**\n\n**LOS ANGELES, CA 90027**\n\n**Consignee's Account Number**\n\n**A13**\n\n**2029 N BERENDO ST**\n\n**LOS ANGELES, CA 90027**\n\n**A13**\n\n**2029 N BERENDO ST**\n\n**LOS ANGELES, CA 90027**\n\n**Consignee's Account Number**\n\n**A13**\n\n**2029 N BERENDO ST**\n\n**LOS ANGELES, CA 90027**\n\n**Accounting Information**\n\n**FREIGHT COLLECT**\n\n|  IATA Code | Account No.  |\n| --- | --- |\n|  |   |\n\n**Place/Airport of Departure and requested Routing**\n\n**SHENZHEN, CHINA**\n\n|  IATA Code | Routing and Destination  |\n| --- | --- |\n|  037001 | 2

pages=[OCRPageObject(index=0, markdown="# 921-41740101\n\n**AASZX2206288**\n\n**SHANGHAI TONGRUO CLOTHING CO., LTD**\n\n**BUILDING 9, NO.199, CHANGJIAN ROAD, BAOSHAN**\n\n**DISTRICT, SHANGHAI, CHINA**\n\n**CHRISTINE HUANG 13501692262**\n\n**AWET**\n\nAs the Carrier\n\nCopies 1, 2 and 3 of this Air Waybill are originals and have the same validity.\n\n**Consignee's Name and Address**\n\n**TA3 SWIM (FUTURE WOW, INC)**\n\n**2029 N BERENDO ST**\n\n**LOS ANGELES, CA 90027**\n\n**Consignee's Account Number**\n\n**A13**\n\n**2029 N BERENDO ST**\n\n**LOS ANGELES, CA 90027**\n\n**A13**\n\n**2029 N BERENDO ST**\n\n**LOS ANGELES, CA 90027**\n\n**Consignee's Account Number**\n\n**A13**\n\n**2029 N BERENDO ST**\n\n**LOS ANGELES, CA 90027**\n\n**Accounting Information**\n\n**FREIGHT COLLECT**\n\n|  IATA Code | Account No.  |\n| --- | --- |\n|  |   |\n\n**Place/Airport of Departure and requested Routing**\n\n**SHENZHEN, CHINA**\n\n|  IATA Code | Routing and Destination  |\n| --- | --- |\n|  037001 | 2022 - 07 - 04  |\n\n**Place/Airport of Destination**\n\n**Fight/Date**\n\n**Fight/Date**\n\n**Fight/Date**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n**A13**\n\n**2022 - 07 - 04**\n\n", images=[], dimensions=OCRPageDimensions(dpi=200, height=2339, width=1654))] model='mistral-ocr-2505-completion' usage_info=OCRUsageInfo(pages_processed=1, doc_size_bytes=256219) document_annotation=None